### ISSUE TITLE
Display purchase price on open positions

### ISSUE THREAD

**Issue by dpaiton** [delvtech/hyperdrive-frontend#291]
**Is your feature request related to a problem? Please describe.**
I just checked the site after spending some time away, and noticed that the fixed rate is low. I went to check on my existing long purchase to see if 
1) it would make sense to close it now (I'm presently wondering if there is ever an advantage to closing early, but that's a separate discussion)
2) how much I would make/lose if I closed it now
The open position has this information:
<img width="1063" alt="image" src="https://github.com/delvtech/hyperdrive-monorepo/assets/153468/da83154b-7c6d-4e4e-97e0-abf787760afc">
So I can see how many bonds I have and the current value, but I can't see how much I paid. So I can't easily calculate my profit or loss.

**Describe the solution you'd like**
If you saved & displayed the total purchase price for the bonds in a given group then you could know how much you've made or lost.

**Describe alternatives you've considered**
TL;DR -- If you're grouping Longs by checkpoint time (or maturity time, it's the same group) then you should list the total amount spent in that group (which means saving every price at time of purchase). Displaying the average price or the price at the corresponding checkpoint time are both potentially misleading.

-----

I might be way off here, but I think there's a challenge that comes from how you're grouping the Longs. I'm going to be verbose with this to help organize my thoughts.

I assume you're grouping long positions by "fungibility", which would mean by checkpoint time (probably by maturity time, which is checkpoint time + position duration). At least, that's how we group them. Checkpointing pushes all open positions within a certain window to the front of that window (e.g. one could configure it such that all positions purchased in a day have the same "mint time" set to the beginning of that day). The discrepancy between spot price at time of purchase & spot price at the checkpoint time is resolved by over-charging the user at the time of purchase proportional to the time gap, so that all purchases within a checkpoint get the same reimbursement, which "reimburses" the over-charged amount. When they close their position (which is based on the checkpoint time) their _profit_ is unchanged -- the magnitude is just shifted depending on when trade occurred in the checkpoint interval. So if we bucket longs by checkpoint time then we lose the individual purchase price.

All of this is to say that, while you may be able to record the purchase price per Long trade, you'd have to display every one individually to reflect it. Since you're grouping purchases by maturity time, I think the right move is to display the total amount spent within the group. I originally thought average, and then thought checkpoint time, before realizing total is the number that most accurately answers the question "how much am I gonna make?".

**Additional context**
Here's another picture of Ramsey, my dog. He was sleeping in pillows, but at the time of the picture he is stressed because he hates getting his picture taken about as much as he loves sleeping in pillows.
![1000000705](https://github.com/delvtech/hyperdrive-monorepo/assets/153468/3a33a869-b0dd-4c00-b428-d7f7a57f193d)
